catalyst-gateway/event-db/Earthfile

```diff
@@ -4,6 +4,26 @@
 
 VERSION 0.7
 
+# cspell: words pgdg eventdb dbname 
```

Comment:
Create a base postgres container, which is based on the latest alpine image.  I know we were using Postgres 14 in cat-core, but let's use this opportunity to upgrade so make a base postgres image like this:

```Earthly 
postgress-base:
    FROM base-postgress:16.0-alpine3.18

    WORKDIR /root

    # Install necessary packages 
    # Expand this list as needed, rather than adding more tools in later containers.
     RUN apk add --no-cache \
            ripgrep \
            bash \
            colordiff
```

You can see what the official postgres container for alpine has in it here:
https://github.com/docker-library/postgres/blob/6f4ae836406b010948f01fbcb400a31dca4fdf52/16/alpine3.18/Dockerfile
So no need to add anything it has.

The reason for this is it will align us with best practice for postgres from the upstream, help lock our version and also let us tweak the container to add some extra tools.

For example, we can add the `dbviz` tool into this and then we can either expose it as a artifact, or use it in scripting when we run stuff inside this container.

Alpine is (as I have learned) what we should target whenever possible as it will help us enforce static builds.  Not that its a huge deal for postgres, but its certainly worth keeping consistent across the CI stack, and I am sure this will probably be part of our "standards" doc for setting up CI.

Reply:
Note, I add:
 `bash` because by default alpine doesn't use it and we need it for better/consistent scripting when required.
`ripgrep` because alpine `grep` is cut down and missing features, and `rg` is a better replacement with all the grep features we need.
`colordiff` we should use instead of diff, if we need to show changes during a CI run because its nicer to review the CI logs when they are colorized.

All of these are pretty small, relatively speaking.
If when your adding steps, a step breaks because of a missing tool, don't add the tool there just add it to this list and rebuild the base container.  This way we have one place where we are maintaining the tools we add to our postgress tooling, not many spread all over the place.